src/bindgen/language_backend/clike.rs

```diff
@@ -911,30 +911,42 @@ impl LanguageBackend for CLikeLanguageBackend<'_> {
                 fields,
                 path,
             } => {
+                let is_constexpr = self.config.language == Language::Cxx
+                    && (self.config.constant.allow_constexpr || l.can_be_constexpr());
                 if self.config.language == Language::C {
                     write!(out, "({})", export_name);
                 } else {
                     write!(out, "{}", export_name);
                 }
 
                 write!(out, "{{ ");
-                let mut is_first_field = true;
                 // In C++, same order as defined is required.
                 let ordered_fields = out.bindings().struct_field_names(path);
-                for ordered_key in ordered_fields.iter() {
+                for (i, ordered_key) in ordered_fields.iter().enumerate() {
                     if let Some(lit) = fields.get(ordered_key) {
-                        if !is_first_field {
-                            write!(out, ", ");
-                        }
-                        is_first_field = false;
-                        if self.config.language == Language::Cxx {
+                        if is_constexpr {
+                            if i > 0 {
+                                write!(out, ", ");
+                            }
```

Comment:
Factor this out of the if/else?

Reply:
If you looked in first commit, this is changed in second commit

Reply:
If you don't mind, I separated the former 2 commits to another PR in #988, which is not directly related to cfg handling.